src/twister2/twister_config.py

```diff
@@ -102,3 +120,38 @@ def get_platform(self, name: str) -> PlatformSpecification:
             if platform.identifier == name:
                 return platform
         raise KeyError(f'There is not platform with identifier: {name}')
+
+
+def _get_selected_platforms(config: pytest.Config) -> list[str]:
+    """Return list of selected platforms"""
+    platforms: list[PlatformSpecification] = config._platforms  # type: ignore
+    emulation_only: bool = config.option.emulation_only
+    architectures: list[str] = config.option.arch
+    all_filter: bool = config.option.all
+    platform_filter: list[str] = config.option.platform
+
+    selected_platforms: list[str] = []
+    if platform_filter:
+        # TODO: verify_platforms_existence
+        selected_platforms = list(set(platform_filter))
+    elif emulation_only:
+        selected_platforms = [
+            platform.identifier for platform in platforms
+            if platform.simulation != 'na'
+        ]
+    elif architectures:
+        selected_platforms = [
+            platform.identifier for platform in platforms
+            if platform.arch in architectures
+        ]
+    elif all_filter:
+        selected_platforms = [
+            platform.identifier for platform in platforms
+        ]
+    else:
+        selected_platforms = [
+            platform.identifier for platform in platforms
+            if platform.testing.default
+        ]
```

Comment:
I think this part from v1 is missing: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/pylib/twister/twisterlib/testplan.py#L578

            # default platforms that can't be run are dropped from the list of
            # the default platforms list. Default platforms should always be
            # runnable.
            for p in _platforms:
                if p.simulation and p.simulation_exec:
                    if shutil.which(p.simulation_exec):
                        platforms.append(p)
                else:
                    platforms.append(p)

Reply:
It will be added  in next phase, when preparing platform scope - the difference with TwisterV1 will be, that we mark this scenario/platform as skipped with proper reason, instead of silently discarding platform.